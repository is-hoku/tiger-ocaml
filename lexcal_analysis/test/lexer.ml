open Lexcal_analysis

let rec tokenize buf =
  match Lexer.read buf with Token.EOF -> [] | t -> t :: tokenize buf

let print_token ch =
  ch |> Lexing.from_channel |> tokenize |> List.map Token.to_string
  |> String.concat " " |> print_string

let () =
  let fin = open_in "../../example_program/test1.tig" in
  try print_token fin with
  | Lexer.ParseError e ->
      print_endline ("Error occurred in lexer: " ^ e);
      close_in fin
  | Lexer.SyntaxError e ->
      print_endline ("Error occurred in lexer: " ^ e);
      close_in fin
