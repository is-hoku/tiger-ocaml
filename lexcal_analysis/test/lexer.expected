let type id(arrtype) = array of id(int) var id(arr1) : id(arrtype) := id(arrtype) [ 10 ] of 0 in id(arr1) end